presto-array/src/main/java/com/facebook/presto/array/ReferenceCountMap.java

```diff
@@ -87,6 +88,9 @@ else if (key instanceof Block) {
         else if (key instanceof Slice) {
             extraIdentity = (int) ((Slice) key).getRetainedSize();
         }
+        else if (key instanceof SliceOutput) {
+            extraIdentity = (int) ((SliceOutput) key).getRetainedSize();
```

Comment:
`SliceOutput` retained size changes in time (it could be close to 0 at first call) while other type of keys are static in size. What is your use case? Why do you need to could references of block builder?

Reply:
> I might return a BlockBuilder instead of Blocks when creating a Page

Could you `build` a page before returning it. Block builders are mutable and it's anti pattern for them to leak outside of connectors.

Reply:
Actually we have a custom block which wraps this block builder (similar to DictionaryBlock) and it is immutable. We are not directly exposing block builders outside connectors.